### DISCUSSION TITLE
Load/restore model without rebuilding data and network object?

### DISCUSSION THREAD

**Discussion by Nhatkhang** [lululxvi/deepxde#1857] [Q&A]
Hi all, 
I am encountering a bottle neck when restore a pre-trained model for a new set of data (the data only have input for prediction). However, when I tried to restored the model from a pre-trained model, its requires data object and net object. 
The data object is something looks like: `data = CartesianProd(x_train, y_train, x_test, y_test)`. Obviously, there is no `y_train, x_test, y_test `available here. How can I resolve this issue without loading the training data to create the same data object, and is there a way to prevent this loading every time a new set of data is inputed?

Thank you in advance.

**Comment by vl-dud** [comment]
I use fake data for such case:
```
# generate CartesianProd that wraps numpy arrays
# array sizes depend on net input/output
data = generate_fake_data(net)

model = dde.Model(data, net)
model.compile()
model.restore(checkpoint_path)
```
This is just my solution, not sure if this is the best way.
@lululxvi Could you please comment on this?

**Reply by Nhatkhang** [reply]
Is the generate_fake_data avaible in deepxde library or just a function to create similar data shape?

**Reply by vl-dud** [reply]
Just my custom function

**Reply by lululxvi** [reply]
Yes, just use some fake data, as the data will not be used if you don't train the model.
